### ISSUE TITLE
Question about mask

### ISSUE THREAD

**Issue by cyysc1998** [OpenGVLab/Diffree#13]
Thanks for sharing the great work! I have a question about the predicted mask. It seems that module OMP and SD are trained separately, and the predicted mask is not involved in the image generation process. So the mask only plays a role in positioning after generation?

**Comment by Lirui-Zhao** [comment]
Hi,

Thank you for your question!

Regarding the training of OMP and SD, while their loss functions are separate, the inputs for both remain consistent, allowing them to be trained simultaneously without issue.

For the mask from OMP, as illustrated in Fig. 7 of the [Diffree paper](https://arxiv.org/pdf/2407.16982), it can be output in the initial decoding step rather than after generation, due to the training process mentioned above. The mask provides crucial location and shape information of the expected added object. As shown in Fig. 10 of the paper, this information is utilized for mixing after generation to ensure background consistency, particularly in iterative additions. The mixing operation is common in mask-based methods (since the input includes a mask), but it poses challenges for implementation in prior text-based methods.

Additionally, The instance mask can be further combined with various existing works to develop exciting applications, as depicted in Fig. 11, with a less time consumption by eliminating the need to wait for the entire generation process.

I hope this helps clarify your question!

Best,
Lirui Zhao

**Comment by cyysc1998** [comment]
Thank you for your reply!
